lib/runner.coffee

```diff
@@ -95,21 +101,24 @@ class Runner
       arg = arg.replace(/{FILE_ACTIVE_NAME_BASE}/g, path.join(codeContext.filename, '..'))
     if project_path?
       arg = arg.replace(/{PROJECT_PATH}/g, project_path)
-    
+
     arg
 
   args: (codeContext, extraArgs) ->
     args = (@scriptOptions.cmdArgs.concat extraArgs).concat @scriptOptions.scriptArgs
-    project_path = ''
-    paths = atom.project.getPaths()
-    if paths.length > 0
-      fs.stat(paths[0], (err, stats) ->
-        if !err
-          project_path = if stats.isDirectory() then paths[0] else path.join(paths[0], '..')
-      )
-    
+    project_path = @getProjectPath or ''
     args = (@fillVarsInArg arg, codeContext, project_path for arg in args)
-    
+
     if not @scriptOptions.cmd? or @scriptOptions.cmd is ''
       args = codeContext.shebangCommandArgs().concat args
     args
+
+  getProjectPath: ->
```

Comment:
Did some further testing. Thought that I might have had a variable scope issue (per http://stackoverflow.com/questions/10810815/variable-scope-in-coffeescript-for-loop), but that was not the case. The issue is that `getProjectPath` finishes executing before the `fs.stats()` callback is finished. I know this because I placed several `console.log`s in the code and the log from the callback always gets printed last.

In the case of master, it seems it's just a matter of luck that the callback finishes and sets `project_path` before the value gets used.

My conclusion is this: using `fs.stat()` instead of `fs.statSync()` breaks the functionality that I am trying to add.